Brokerages/Oanda/OandaBrokerage.cs

```diff
@@ -0,0 +1,1052 @@
+        {
+            var handler = SessionRefreshed;
+            if (handler != null) handler(this, e);
+        }
+
+        /// <summary>
+        /// Connects the client to the broker's remote servers
+        /// </summary>
+        public override void Connect()
+        {
+            if (IsConnected) return;
+            RefreshSession();
+        }
+
+        /// <summary>
+        /// Disconnects the client from the broker's remote servers
+        /// </summary>
+        public override void Disconnect()
+        {
+            throw new NotImplementedException();
+        }
+
+        /// <summary>
+        /// Event fired when our session has been refreshed/tokens updated
+        /// </summary>
+        public event EventHandler<TokenResponse> SessionRefreshed;
+
+        /// <summary>
+        /// Set the access token and login information for the Oanda brokerage 
+        /// </summary>
+        /// <param name="userId">Userid for this brokerage</param>
+        /// <param name="accessToken">Viable access token</param>
+        /// <param name="refreshToken">Our refresh token</param>
+        /// <param name="issuedAt">When the token was issued</param>
+        /// <param name="lifeSpan">Life span for our token.</param>
+        public void SetTokens(int userId, string accessToken, string refreshToken, DateTime issuedAt, TimeSpan lifeSpan)
+        {
+            AccessToken = accessToken;
+            RefreshToken = refreshToken;
+            _issuedAt = issuedAt;
+            _lifeSpan = lifeSpan;
+            CheckForFills();
+        }
+
+        private static Stream GetResponseStream(WebResponse response)
+        {
+            var stream = response.GetResponseStream();
+            if (response.Headers["Content-Encoding"] == "gzip")
+            {	// if we received a gzipped response, handle that
+                if (stream != null) stream = new GZipStream(stream, CompressionMode.Decompress);
+            }
+            return stream;
+        }
+
+        /// <summary>
+        /// Initializes a streaming events session which will stream events for the given accounts
+        /// </summary>
+        /// <param name="accountId">the account IDs you want to stream on</param>
+        /// <returns>the WebResponse object that can be used to retrieve the events as they stream</returns>
+        public static async Task<WebResponse> StartEventsSession(List<int> accountId = null)
+        {
+            var requestString = EndpointResolver.ResolveEndpoint(OandaEnvironment, Server.StreamingEvents) + "events";
+
+            if (accountId != null && accountId.Count > 0)
+            {
+                var accountIds = accountId.Aggregate("", (current, account) => current + (account + ","));
+                accountIds = accountIds.Trim(',');
+                requestString += "?accountIds=" + WebUtility.UrlEncode(accountIds);
+            }
+
+            var request = WebRequest.CreateHttp(requestString);
```

Comment:
in all my years with C# (~10) I have not seen `WebRequest.CreateHttp` thank you for showing me this, I've always wondered why someone would design an API that requires casting! Typically I see:
```c#
var http = (HttpWebRequest)WebRequest.Create(url);
```

and always wondered but never investigated if it could be done without the cast, sweet! learn something new every day.